unitc

```diff
@@ -9,39 +9,49 @@ METHOD=PUT
 SHOW_LOG=1
 URI=""
 
-while [ $# -gt 0 ]; do
-	OPTION=$(echo $1 | tr '[a-z]' '[A-Z]')
-	case $OPTION in
-		"-H" | "--HELP")
-			shift
-			;;
-
-		"-Q" | "--QUIET")
-			QUIET=1
-			shift
-			;;
-
-		"GET" | "PUT" | "POST" | "DELETE" | "INSERT")
-			METHOD=$OPTION
-			shift
-			;;
-
-		"HEAD" | "PATCH" | "PURGE" | "OPTIONS")
-			echo "${0##*/}: ERROR: Invalid HTTP method ($OPTION)"
-			exit 1
-			;;
-
-		*)
-			if [ "${1:0:1}" = "/" ]; then
-				URI=$1
-				shift
-			else
-				echo "${0##*/}: ERROR: Invalid option ($1)"
-				exit 1
-			fi
-			;;
-	esac
+usage() {
+	echo "-h: This message"
+	echo "-m: <method> GET / PUT / POST / DELETE / INSERT"
+	echo "-u: <uri> URI or unit config path"
+}
+
+check_unit_running() {
+  # Check if Unit is running, find the main process
+  #
+  PID=($(ps ax | grep unit:\ main | grep -v \ grep | awk '{print $1}'))
+  if [ ${#PID[@]} -eq 0 ]; then
+  	echo "${0##*/}: ERROR: unitd not running"
+  	exit 1
+  elif [ ${#PID[@]} -gt 1 ]; then
+  	echo "${0##*/}: ERROR: multiple unitd processes detected (${PID[@]})"
+  	exit 1
+  fi
+}
+
+while getopts "m:u:h" o; do
+    case "${o}" in
+        h)
+            usage
+            ;;
+        m)
+            METHOD=$(echo ${OPTARG} | tr '[a-z]' '[A-Z]')
+            ;;
+	u)
+	    URI=${OPTARG}
+	    ;;
+        *)
+            usage
+            ;;
+    esac
 done
+shift $((OPTIND-1))
```

Comment:
While I like using standard or semi-standard ways to do things, we discussed internally that we want it to be both (1) portable and (2) accept long options.

None of the existing tools is portable enough while allowing both long options and short options.

In my competing client tool for unitd(8), which you may also want to check, I did it this way:

<https://github.com/alejandro-colomar/unitcli/blob/main/bin/setup-unit#L198>

I wrote it to have the features of `unitc`, but also be more modular and have more functionality than `unitc`.  However, it means that it's also slightly more uncomfortable to use (you have to pass a few more levels of subcommands for the same functionality.  You may want to check any of the repos here:

<http://www.alejandro-colomar.es/src/alx/nginx/unitcli.git/>
<https://github.com/alejandro-colomar/unitcli/>